tools/distpartitioning/globalids.py

```diff
@@ -50,7 +50,7 @@ def get_shuffle_global_nids(rank, world_size, global_nids_ranks, node_data):
     #form the outgoing message
     send_nodes = []
     for i in range(world_size):
-        send_nodes.append(torch.Tensor(global_nids_ranks[i]).type(dtype=torch.int64))
+        send_nodes.append(torch.from_numpy(global_nids_ranks[i]).type(dtype=torch.int64))
```

Comment:
is it related to the bug?

Reply:
Yes. This is the bug on which I apparently spent a week debugging with the mag_lsc dataset. Originally the tensor was created, with global_nids, using "torch.Tensor()" function call. Please note that the argument here global_nids_ranks[i] is a numpy array. This works for smaller datasets where this list is on the smaller side and I have not noticed any mem. corruption issues. But when testing with mag_lsc dataset, right after this function call when we compare the actual tensor created vs. the data passed to this function call ... they are not identical. There was some wierd kind of mem. corruption hapenning when the input numpy array is very large and torch.Tensor() was used in creating the Tensor. 

This issue seems to be resolved, when torch.from_numpy() function call was used to create the Tensor. In this case for larger messages the data in the newly created tensor and the actual data passed to this function call are always correct.